hoodie-common/src/main/java/com/uber/hoodie/common/table/TableFileSystemView.java

```diff
@@ -0,0 +1,93 @@
+/*
+ *  Copyright (c) 2016 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.uber.hoodie.common.table;
+
+import com.uber.hoodie.common.model.HoodieDataFile;
+import com.uber.hoodie.common.model.HoodieRecord;
+import org.apache.hadoop.fs.FileStatus;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+/**
+ * Interface for viewing the table file system.
+ * Dependening on the Hoodie Table Type - The view of the filesystem changes.
+ * <p>
+ * ReadOptimizedView - Lets queries run only on organized columnar data files at the expense of latency
+ * WriteOptimizedView - Lets queries run on columnar data as well as delta files (sequential) at the expense of query execution time
+ * @since 0.3.0
+ */
+public interface TableFileSystemView {
+    /**
+     * Stream all the data files for a specific FileId.
+     * This usually has a single RO file and multiple WO files if present.
+     *
+     * @param partitionPath
+     * @param fileId
+     * @return
+     */
+    Stream<HoodieDataFile> getLatestDataFilesForFileId(final String partitionPath,
+        final String fileId);
+
+    /**
+     * Stream all the latest version data files in the given partition
+     * with precondition that commitTime(file) before maxCommitTime
+     *
+     * @param partitionPathStr
+     * @param maxCommitTime
+     * @return
+     */
+    Stream<HoodieDataFile> streamLatestVersionInPartition(String partitionPathStr,
```

Comment:
`streamLatest...` is a bit confusing, since its just doing some metadata fetches.. Ideally we reserve such terminology for actual data streaming.. Can we rename to `getLatestVersionInPartition` and so forth here and also below?